src/analysis/retail/monk/mistweaver/modules/spells/CloudedFocus.tsx

```diff
@@ -128,32 +122,52 @@ class CloudedFocus extends Analyzer {
     this.addEventListener(
       Events.heal
         .by(SELECTED_PLAYER)
-        .spell([SPELLS.VIVIFY, TALENTS_MONK.ENVELOPING_MIST_TALENT, SPELLS.ENVELOPING_BREATH_HEAL]),
+        .spell([
+          SPELLS.VIVIFY,
+          SPELLS.INVIGORATING_MISTS_HEAL,
+          TALENTS_MONK.ENVELOPING_MIST_TALENT,
+          SPELLS.ENVELOPING_BREATH_HEAL,
+        ]),
       this.calculateHealingEffect,
     );
   }
 
   calculateManaEffect(event: CastEvent) {
     const spellId = event.ability.guid;
-    if (spellId === SPELLS.VIVIFY.id && this.stacks > 0) {
-      this.tallyPrimaryTargetOverheal(event);
-    }
     if (this.selectedCombatant.hasBuff(SPELLS.INNERVATE.id)) {
       return;
     }
-    debug && console.log('Current Stacks: ', this.stacks, 'Mana Stacks: ', this.manaStacks, event);
 
     const cost = event.resourceCost ? event.resourceCost[RESOURCE_TYPES.MANA.id] : 0;
-    this.addManaToStackMap(spellId, cost);
-    const preCFCost = cost / (1 - CF_BUFF_PER_STACK * this.manaStacks);
-    this.manaSaved += preCFCost - cost;
+    if (cost === 0) {
+      debug && console.log('No Mana Cost Found: ', event);
+    }
+
+    const manaSaved = cost / (1 - CF_BUFF_PER_STACK * this.stacks) - cost;
+    this.addManaToStackMap(spellId, manaSaved);
+    this.manaSaved += manaSaved;
+
+    debug &&
+      console.log(
+        'Mana: ',
+        'Cost - ',
+        cost,
+        'Saved - ',
+        manaSaved,
+        'Stacks - ',
+        this.stacks,
+        this.owner.formatTimestamp(event.timestamp),
+      );
   }
 
   calculateHealingEffect(event: HealEvent) {
     const spellId = event.ability.guid;
     if (this.stacks === 0) {
       return;
     }
+    if (spellId === SPELLS.VIVIFY.id && this.stacks > 0) {
```

Comment:
```suggestion
    if (spellId === SPELLS.VIVIFY.id) {
```
We already return early, no need